File: Turquoise/UI/Common/TQLabel.m

#import "TQLabel.h"

static NSString *const kHorizontalInsetKey = @"horizontalInset";
static NSString *const kVerticalInsetKey = @"verticalInset";

@implementation TQLabel

- (instancetype)initWithCoder:(NSCoder *)aDecoder {
  self = [super initWithCoder:aDecoder];
  if (self) {
    CGFloat fontSize = self.font.pointSize;
    self.font = [UIFont fontWithName:@"dungeon" size:fontSize];
  }

  return self;
}

- (void)setHorizontalInset:(CGFloat)horizontalInset {
  if (_horizontalInset != horizontalInset) {
    _horizontalInset = horizontalInset;
    [self setNeedsLayout];
  }
}

- (void)setVerticalInset:(CGFloat)verticalInset {
  if (_verticalInset != verticalInset) {
    _verticalInset = verticalInset;
    [self setNeedsLayout];
  }
}

#pragma mark - UILabel Overrides

- (void)drawTextInRect:(CGRect)rect {
  // top, left, bottom, right
  UIEdgeInsets insets = {_verticalInset, _horizontalInset, _verticalInset, _horizontalInset};
  [super drawTextInRect:UIEdgeInsetsInsetRect(rect, insets)];
}

@end

File: Turquoise/UI/Common/TQLabel.swift
import Foundation
import UIKit

class TQLabel : UILabel {
  var horizontalInset: CGFloat = 0 {
    didSet {
      setNeedsLayout()
    }
  }

  var verticalInset: CGFloat = 0 {
    didSet {
      self.setNeedsLayout()
    }
  }

    var fontSize: CGFloat {
        get {
            return self.font.pointSize
        }
        set {
            self.font = UIFont(name: "dungeon", size: newValue)
        }

    }

    override init(frame: CGRect) {
        super.init(frame: frame)
        self.font = UIFont(name: "dungeon", size: self.font.pointSize)
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    override func drawText(in rect: CGRect) {
        let insets = UIEdgeInsets(top: verticalInset, left: horizontalInset, bottom: verticalInset, right: horizontalInset)
        super.drawText(in: rect.inset(by: insets))
    }
}
